File: PayrollProjectNew/payroll_app/lib/src/features/advance/model/advance.dart
import 'package:payroll_app/src/features/employee/model/employee.dart';

class Advance {
  int? id;
  Employee? employee;
  double? amount;
  String? reason;
  String? date;

  Advance({this.id, this.employee, this.amount, this.reason, this.date});

  Advance.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    employee = json['employee'] != null
        ? Employee.fromJson(json['employee'])
        : null;
    amount = json['amount'];
    reason = json['reason'];
    date = json['date'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    if (this.employee != null) {
      data['employee'] = this.employee!.toJson();
    }
    data['amount'] = this.amount;
    data['reason'] = this.reason;
    data['date'] = this.date;
    return data;
  }
}

File: Backend/CreateApi/src/main/java/com/shohab/CreateApi/model/Advance.java
package com.shohab.CreateApi.model;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data // Generates getters, setters, toString, equals, and hashCode methods
@NoArgsConstructor // Generates a no-argument constructor
@AllArgsConstructor
@Entity
public class Advance {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    private Employee employee;
    private BigDecimal amount;
    private String reason;
    private String date;

    public Advance(Employee employee, BigDecimal amount) {
        this.employee = employee;
        this.amount = amount;
    }
}
